lsp-mode.el

```diff
@@ -6734,28 +6734,42 @@ should return the command to start the LS server."
   "Create LSP stdio connection named name.
 LOCAL-COMMAND is either list of strings, string or function which
 returns the command to execute."
+  ;; 2.5.0-pre (as built from native-comp branch before M Albinus released tramp-2.5)
+  ;; worked fine
+  (defvar tramp-version)
+  (defvar tramp-connection-properties)
+  (when (version< tramp-version "2.5.0-pre")
+    (lsp-warn
+     "Your tramp version - %s - might fail to work with remote LSP. Update to version 2.5 or greater (available on elpa)"
+     tramp-version))
+  ;; Force a direct asynchronous process.
+  (add-to-list 'tramp-connection-properties
+               (list (regexp-quote (file-remote-p default-directory))
+                     "direct-async-process" t))
   (list :connect (lambda (filter sentinel name environment-fn)
-                   (let* ((final-command (lsp-resolve-final-function local-command))
-                          ;; wrap with stty to disable converting \r to \n
+                   (let* ((final-command (lsp-resolve-final-function
+                                          local-command))
+                          (stderr-buf (or (when generate-error-file-fn
```

Comment:
if stderr buffer works for tramp 2.5 then buffer name can be generated like this `(stderr-buf (format "*%s::stderr*" process-name))`. This is what we do for normal stdio.

Reply:
It complains that stderr-buf does not exist so (stderr-buf (generate-new-buffer stderr-buf)) worked.

Reply:
Something like this seems to work:
```elisp
:stderr (get-buffer-create (format "*%s::stderr*" process-name))
```